src/modules/aws_vpc/mappers/peering_connection.ts

```diff
@@ -0,0 +1,290 @@
+  }
+
+  private convertTagsForAws(tags: { [key: string]: string }) {
+    return Object.keys(tags).map(k => {
+      return {
+        Key: k,
+        Value: tags[k],
+      };
+    });
+  }
+
+  private async getVpcByIdAndRegion(ctx: Context, vpcId: string, region: string) {
+    return (
+      (await this.module.vpc.db.read(ctx, this.module.vpc.generateId({ vpcId, region }))) ??
+      (await this.module.vpc.cloud.read(ctx, this.module.vpc.generateId({ vpcId, region })))
+    );
+  }
+
+  cloud: Crud2<PeeringConnection> = new Crud2({
+    create: async (es: PeeringConnection[], ctx: Context) => {
+      const out = [];
+      for (const e of es) {
+        const region = e.requester.region;
+        const client = (await ctx.getAwsClient(region)) as AWS;
+
+        const input: CreateVpcPeeringConnectionCommandInput = {
+          VpcId: e.requester.vpcId,
+          PeerVpcId: e.accepter.vpcId,
+          PeerRegion: e.accepter.region,
+        };
+        if (e.tags !== undefined && e.tags !== null) {
+          input.TagSpecifications = [
+            {
+              ResourceType: 'vpc-peering-connection',
+              Tags: this.convertTagsForAws(e.tags),
+            },
+          ];
+        }
+
+        const res: AwsPeeringConnection | undefined = await this.createPeeringConnection(
+          client.ec2client,
+          (
+            await ctx.getAwsClient(e.accepter.region)
+          ).ec2client,
+          input,
+        );
+        if (!res || !res.VpcPeeringConnectionId) continue;
+        await this.createTwoWayRoutes(e.requester, e.accepter, res.VpcPeeringConnectionId, ctx);
+
+        const newPeeringConnection = await this.peeringConnectionMapper(res, ctx);
+        if (!newPeeringConnection) continue;
+
+        if (e.id) {
+          newPeeringConnection.id = e.id;
+          await this.module.peeringConnection.db.update(newPeeringConnection, ctx);
+        }
+        out.push(newPeeringConnection);
+      }
+
+      return out;
+    },
+    read: async (ctx: Context, id?: string) => {
+      const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
+      const out: PeeringConnection[] = [];
+      await Promise.all(
+        enabledRegions.map(async region => {
+          const client = (await ctx.getAwsClient(region)) as AWS;
+          for (const peeringConnection of await this.getPeeringConnections(client.ec2client)) {
+            if (
+              peeringConnection?.Status?.Code === 'deleted' ||
+              peeringConnection?.Status?.Code === 'deleting'
+            )
+              continue;
+
+            const outPeeringConnection = await this.peeringConnectionMapper(peeringConnection, ctx);
+            if (outPeeringConnection) out.push(outPeeringConnection);
+          }
+        }),
+      );
+
+      if (!!id) {
+        return out.find(pc => pc.peeringConnectionId === id);
+      }
+
+      return out;
+    },
+    update: async (es: PeeringConnection[], ctx: Context) => {
```

Comment:
Would be good to add here the `updateOrReplace` method and return `replace` when the vpcIds are different.